clouddriver-azure/src/test/groovy/com/netflix/spinnaker/clouddriver/azure/resources/loadbalancer/deploy/validators/UpsertAzureLoadBalancerDescriptionValidatorSpec.groovy

```diff
@@ -0,0 +1,156 @@
+  private static final LOAD_BALANCER_NAME = "azureapp1-st1-d1"
+  private static final REGION = "westus"
+  private static final ACCOUNT_NAME = "azurecred1"
+  private static final CLOUD_PROVIDER = "azure"
+  private static final ACCOUNT_CLIENTID = "azureclientid"
+  private static final ACCOUNT_TENANTID = "azuretenantid1"
+  private static final ACCOUNT_APPKEY = "azureappkey1"
+  private static final APP_NAME = "azureapp1"
+  private static final STACK = "st1"
+  private static final DETAIL = "d1"
+  private static final VNET = "vnet-westus-azureapp1-st1-d1"
+  private static final SECURITY_GROUPS = "azureapp1-sg1-d1"
+  private static final PROBE_NAME1 = "healthcheck1"
+  private static final PROBE_PROTOCOL1 = "HTTP"
+  private static final PROBE_PORT1 = 7001
+  private static final PROBE_PATH1 = "/healthcheck"
+  private static final PROBE_INTERVAL1 = 10
+  private static final PROBE_UNHEALTHY_THRESHOLD1 = 2
+  private static final PROBE_NAME2 = "healthcheck2"
+  private static final PROBE_PROTOCOL2 = "TCP"
+  private static final PROBE_PORT2 = 7002
+  private static final PROBE_INTERVAL2 = 12
+  private static final PROBE_UNHEALTHY_THRESHOLD2 = 4
+  private static final LB_RULE_NAME1 = "lbRule1"
+  private static final LB_RULE_PROTOCOL1 = "TCP"
+  private static final LB_RULE_EXTERNAL_PORT1 = 80
+  private static final LB_RULE_BACKEND_PORT1 = 80
+  private static final LB_RULE_PROBE_NAME1 = "healthcheck1"
+  private static final LB_RULE_IDLE_TIMEOUT1 = 4
+  private static final LB_RULE_NAME2 = "lbRule2"
+  private static final LB_RULE_PROTOCOL2 = "TCP"
+  private static final LB_RULE_EXTERNAL_PORT2 = 8080
+  private static final LB_RULE_BACKEND_PORT2 = 8080
+  private static final LB_RULE_PROBE_NAME2 = "healthcheck2"
+  private static final LB_RULE_IDLE_TIMEOUT2 = 5
+
+  @Shared
+  ObjectMapper mapper = new ObjectMapper()
+
+  @Shared
+  AzureCredentials azureCredentials
+
+  @Shared
+  UpsertAzureLoadBalancerAtomicOperationConverter converter
+
+  @Shared
+  UpsertAzureLoadBalancerDescriptionValidator validator
+
+  void setupSpec() {
+    azureCredentials = new AzureCredentials(ACCOUNT_CLIENTID, ACCOUNT_TENANTID, ACCOUNT_APPKEY, null, null, null)
+
+    def credentialsRepo = new MapBackedAccountCredentialsRepository()
+    def credentials = Mock(AzureNamedAccountCredentials)
+    credentials.getAccountName() >> ACCOUNT_NAME
+    credentials.getName() >> ACCOUNT_NAME
+    credentials.getCredentials() >> azureCredentials
+    credentialsRepo.save(ACCOUNT_NAME, credentials)
+    def credentialsProvider = new DefaultAccountCredentialsProvider(credentialsRepo)
+    validator = new UpsertAzureLoadBalancerDescriptionValidator()
+    validator.accountCredentialsProvider = credentialsProvider
+    this.converter = new UpsertAzureLoadBalancerAtomicOperationConverter(objectMapper: mapper)
+    converter.accountCredentialsProvider = credentialsProvider
+  }
+
+  void "pass validation with proper description inputs"() {
```

Comment:
Yes, I agree. I was hopping that Haishi's site will take care of that rather then us duplicating that work. For now since we check against a constant set of values anyway I'm more concern with the execution flow (i.e. calling our Azure specific implementation) than the final content of the description object being returned.